### DISCUSSION TITLE
Priority Loads

### DISCUSSION THREAD

**Discussion by BenWinchester** [CLOVER-energy/CLOVER#184] [IAA]
### Issue
> "Not all demands are created equally at all times."
\- Anon.

(From #84 ) Within CLOVER, there are several difference demands which are considered: electric demands (conventional plug-in devices), domestic hot-water demands, cooling demands, clean-water demands and potentially more in the pipeline. Whilst it is impossible to compare between a demand placed on the system for electricity with that for clean water (the use of `Converter` instances is a way to equate the two, but the priority of the two demands is difficult to quantify and is a separate issue), it __is__ possible to consider the difference between the same demand placed on the system at
* different times of day,
* and by different users.

(New) Further, there is no differentiation between the **priority** of certain loads. E.G., in some settings, it may be that the energy system should be prioritising some anchor load, e.g., a health centre, over the demands of businesses or individual households.

### Proposal
The introduction of the notion of **load prioritisation** into CLOVER.

* The only alternative I can think of is not implementing this enhancement.

**Comment by hamishbeath** [comment]
It may not be relevant but for Alix’s thesis I implemented a very simplistic (electric) load prioritisation hack. I will refer back to it to see what I did and we can discuss at some point. I am interested in how this is added!

**Comment by BenWinchester** [comment]
That's sounds really good! I wasn't envisioning us doing this anytime soon (and have tagged it "good masters issue" as it might be good for an MSc project next year?), I more just noticed that it was something I'd mentioned at some point in a doc and figured it should probably be trackable on Github somewhere :smile:

**Comment by BenWinchester** [comment]
A thought on this: a user could be able to create custom-defined priority levels as well as, as per #164, whether a load type can be shifted. E.G.,

- Critical loads, such as vaccine refrigeration,
- Priority loads, such as schools etc.,
- Standard loads, such as household uses,

with, perhaps, a boolean flag for shiftable loads.

**Comment by BenWinchester** [comment]
## General thoughts
This likely has relevance under the new _**productive uses**_ theme, whereby certain additional power can be used for things such as cooling ice (which is effectively a form of energy storage), charging electric-vehicle batteries at swap stations (relevant to some of the work that MSc students are doing within the Energy Access research group) and agri-PV (such as pumping water) or even clean-water demands (such as desalinating using excess power). Introducing a tiered system of demand, whereby certain loads are "met" by the energy system, and other "productive uses" are met instead by excess power rather than being a priority.

@hamishbeath , this might be something worth "issuing" and considering as part of agri-PV?
